Gemfile

```diff
@@ -31,6 +31,7 @@ gem 'jbuilder', '~> 2.5'
 # gem 'bcrypt', '~> 3.1.7'
 gem 'figaro'
 gem 'faraday'
+gem "omniauth-google-oauth2"
```

Comment:
Actually yes. I was trying to hand roll it but there was so much JS that I didnt want to get stuck on it for a day and a half. With this we setup an initializer file that you feed your provider info and keys. Another key was adding a callback on the Google Api Console. Also an important note, you have to run localhost when you sign it with google. It does not allow you to call direct ip addresses.